File: atlas-richie-gateway-service/src/main/resources/client-library/swift/example/Url.swift
// Url.swift
// 业务代码：URL枚举定义示例
//
// Author: richie696
// Version: 1.0
// Since: 2025-11-01

import Foundation

/// 业务代码定义的URL枚举
/// 遵循UrlProtocol协议，实现所有必需的属性
enum Url: UrlProtocol {
    // ========== 认证模块 ==========
    /// 用户登录 - 需要加密，不需要防重复提交（允许快速重试）
    case userLogin
    
    /// 用户注册 - 需要加密和防重复提交
    case userRegister
    
    /// 用户登出
    case userLogout
    
    // ========== 用户模块 ==========
    /// 获取用户信息 - 需要加密
    case userProfile
    
    /// 更新用户信息 - 需要加密和防重复提交
    case userUpdate
    
    // ========== 订单模块 ==========
    /// 订单列表 - 公开数据
    case orderList
    
    /// 提交订单 - 需要加密和防重复提交
    case orderSubmit
    
    /// 取消订单 - 需要防重复提交
    case orderCancel
    
    // ========== 支付模块 ==========
    /// 创建支付 - 最高安全级别
    case paymentCreate
    
    /// 查询支付状态 - 需要加密
    case paymentStatus
    
    // ========== 基础数据 ==========
    /// 菜单列表 - 公开数据
    case menuAll
    
    /// 字典数据 - 公开数据
    case dictList
    
    // ========== 实现UrlProtocol协议 ==========
    
    /// 获取URL路径
    var path: String {
        switch self {
        // 认证模块
        case .userLogin:    return "/api/auth/login"
        case .userRegister: return "/api/auth/register"
        case .userLogout:   return "/api/auth/logout"
        
        // 用户模块
        case .userProfile: return "/api/user/profile"
        case .userUpdate:  return "/api/user/update"
        
        // 订单模块
        case .orderList:    return "/api/order/list"
        case .orderSubmit: return "/api/order/submit"
        case .orderCancel:  return "/api/order/cancel"
        
        // 支付模块
        case .paymentCreate: return "/api/payment/create"
        case .paymentStatus: return "/api/payment/status"
        
        // 基础数据
        case .menuAll:  return "/api/menu/all"
        case .dictList: return "/api/dict/list"
        }
    }
    
    /// 获取HTTP方法
    var method: Method {
        switch self {
        case .userLogin: return .post
        case .userRegister: return .post
        case .userLogout: return .post
        case .userProfile: return .get
        case .userUpdate: return .put
        case .orderList: return .get
        case .orderSubmit: return .post
        case .orderCancel: return .post
        case .paymentCreate: return .post
        case .paymentStatus: return .get
        case .menuAll: return .get
        case .dictList: return .get
        }
    }
    
    /// 是否需要加密
    var needEncryption: Bool {
        switch self {
        case .userLogin: return true
        case .userRegister: return true
        case .userLogout: return false
        case .userProfile: return true
        case .userUpdate: return true
        case .orderList: return false
        case .orderSubmit: return true
        case .orderCancel: return false
        case .paymentCreate: return true
        case .paymentStatus: return true
        case .menuAll: return false
        case .dictList: return false
        }
    }
    
    /// 是否需要防重复提交检查
    var needDuplicateCheck: Bool {
        switch self {
        case .userLogin: return false  // 允许快速重试
        case .userRegister: return true
        case .userLogout: return false
        case .userProfile: return false
        case .userUpdate: return true
        case .orderList: return false
        case .orderSubmit: return true
        case .orderCancel: return true
        case .paymentCreate: return true
        case .paymentStatus: return false
        case .menuAll: return false
        case .dictList: return false
        }
    }
}

// ========== 可选的辅助属性和方法 ==========

extension Url {
    /// 获取枚举名称（用于调试）
    var name: String {
        switch self {
        case .userLogin: return "USER_LOGIN"
        case .userRegister: return "USER_REGISTER"
        case .userLogout: return "USER_LOGOUT"
        case .userProfile: return "USER_PROFILE"
        case .userUpdate: return "USER_UPDATE"
        case .orderList: return "ORDER_LIST"
        case .orderSubmit: return "ORDER_SUBMIT"
        case .orderCancel: return "ORDER_CANCEL"
        case .paymentCreate: return "PAYMENT_CREATE"
        case .paymentStatus: return "PAYMENT_STATUS"
        case .menuAll: return "MENU_ALL"
        case .dictList: return "DICT_LIST"
        }
    }
    
    /// 判断是否为写操作
    var isWriteOperation: Bool {
        switch method {
        case .post, .put, .delete, .patch: return true
        default: return false
        }
    }
}

File: atlas-richie-gateway-service/src/main/resources/client-library/rust/examples/url.rs
//! url.rs
//! 业务代码：URL枚举定义示例
//!
//! Author: richie696
//! Version: 1.0
//! Since: 2025-11-01

use httpclient::{Method, UrlTrait};
use std::fmt;

/// 业务代码定义的URL枚举
/// 实现UrlTrait trait
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
pub enum Url {
    // ========== 认证模块 ==========
    /// 用户登录 - 需要加密，不需要防重复提交（允许快速重试）
    UserLogin,
    
    /// 用户注册 - 需要加密和防重复提交
    UserRegister,
    
    /// 用户登出
    UserLogout,
    
    // ========== 用户模块 ==========
    /// 获取用户信息 - 需要加密
    UserProfile,
    
    /// 更新用户信息 - 需要加密和防重复提交
    UserUpdate,
    
    // ========== 订单模块 ==========
    /// 订单列表 - 公开数据
    OrderList,
    
    /// 提交订单 - 需要加密和防重复提交
    OrderSubmit,
    
    /// 取消订单 - 需要防重复提交
    OrderCancel,
    
    // ========== 支付模块 ==========
    /// 创建支付 - 最高安全级别
    PaymentCreate,
    
    /// 查询支付状态 - 需要加密
    PaymentStatus,
    
    // ========== 基础数据 ==========
    /// 菜单列表 - 公开数据
    MenuAll,
    
    /// 字典数据 - 公开数据
    DictList,
}

// 实现UrlTrait trait
impl UrlTrait for Url {
    /// 获取URL路径
    fn path(&self) -> &'static str {
        match self {
            // 认证模块
            Url::UserLogin => "/api/auth/login",
            Url::UserRegister => "/api/auth/register",
            Url::UserLogout => "/api/auth/logout",
            
            // 用户模块
            Url::UserProfile => "/api/user/profile",
            Url::UserUpdate => "/api/user/update",
            
            // 订单模块
            Url::OrderList => "/api/order/list",
            Url::OrderSubmit => "/api/order/submit",
            Url::OrderCancel => "/api/order/cancel",
            
            // 支付模块
            Url::PaymentCreate => "/api/payment/create",
            Url::PaymentStatus => "/api/payment/status",
            
            // 基础数据
            Url::MenuAll => "/api/menu/all",
            Url::DictList => "/api/dict/list",
        }
    }
    
    /// 获取HTTP方法
    fn method(&self) -> Method {
        match self {
            Url::UserLogin => Method::Post,
            Url::UserRegister => Method::Post,
            Url::UserLogout => Method::Post,
            Url::UserProfile => Method::Get,
            Url::UserUpdate => Method::Put,
            Url::OrderList => Method::Get,
            Url::OrderSubmit => Method::Post,
            Url::OrderCancel => Method::Post,
            Url::PaymentCreate => Method::Post,
            Url::PaymentStatus => Method::Get,
            Url::MenuAll => Method::Get,
            Url::DictList => Method::Get,
        }
    }
    
    /// 是否需要加密
    fn need_encryption(&self) -> bool {
        match self {
            Url::UserLogin => true,
            Url::UserRegister => true,
            Url::UserLogout => false,
            Url::UserProfile => true,
            Url::UserUpdate => true,
            Url::OrderList => false,
            Url::OrderSubmit => true,
            Url::OrderCancel => false,
            Url::PaymentCreate => true,
            Url::PaymentStatus => true,
            Url::MenuAll => false,
            Url::DictList => false,
        }
    }
    
    /// 是否需要防重复提交检查
    fn need_duplicate_check(&self) -> bool {
        match self {
            Url::UserLogin => false,  // 允许快速重试
            Url::UserRegister => true,
            Url::UserLogout => false,
            Url::UserProfile => false,
            Url::UserUpdate => true,
            Url::OrderList => false,
            Url::OrderSubmit => true,
            Url::OrderCancel => true,
            Url::PaymentCreate => true,
            Url::PaymentStatus => false,
            Url::MenuAll => false,
            Url::DictList => false,
        }
    }
}

// 可选的辅助方法
impl Url {
    /// 获取枚举名称（用于调试）
    pub fn name(&self) -> &'static str {
        match self {
            Url::UserLogin => "USER_LOGIN",
            Url::UserRegister => "USER_REGISTER",
            Url::UserLogout => "USER_LOGOUT",
            Url::UserProfile => "USER_PROFILE",
            Url::UserUpdate => "USER_UPDATE",
            Url::OrderList => "ORDER_LIST",
            Url::OrderSubmit => "ORDER_SUBMIT",
            Url::OrderCancel => "ORDER_CANCEL",
            Url::PaymentCreate => "PAYMENT_CREATE",
            Url::PaymentStatus => "PAYMENT_STATUS",
            Url::MenuAll => "MENU_ALL",
            Url::DictList => "DICT_LIST",
        }
    }
    
    /// 判断是否为写操作
    pub fn is_write_operation(&self) -> bool {
        matches!(self.method(), Method::Post | Method::Put | Method::Delete | Method::Patch)
    }
}

impl fmt::Display for Url {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(
            f,
            "Url::{}(path={}, method={}, encryption={}, duplicateCheck={})",
            self.name(),
            self.path(),
            self.method(),
            self.need_encryption(),
            self.need_duplicate_check()
        )
    }
}

